src/spac/data_utils.py

```diff
@@ -8,81 +8,55 @@
 from spac.utils import regex_search_list
 
 
-def append_observation(
+def append_annotation(
     data: pd.DataFrame,
-    source_column: str,
-    new_column: str,
-    mapping_rules: list
+    annotation: list
 ) -> pd.DataFrame:
     """
-    Append a new observation to a Pandas DataFrame based on mapping rules.
+    Append a new annotation with single value to
+    a Pandas DataFrame based on mapping rules.
 
     Parameters
     ----------
     data : pd.DataFrame
         The input DataFrame to which the new observation will be appended.
-    source_column : str
-        The name of the column in the 'data' DataFrame that
-        contains the values to be mapped.
-    new_column : str
-        The name of the new column to be added to the DataFrame.
-    mapping_rules : list
-        A list of strings representing the mapping rules.
-        Each string should have this format:
-        <value in source column>:<value in new observation>
+
+    annotation : list
+        list of dictionary of strings representing
+        the new annotation and its value.
+        Each dictionary should have this format:
+        {<new annotation column name>:<value of the annotation>}
+        The values must be a single string or numeric value.

```

Comment:
@ruiheesi, I believe you can add one dictionary with multiple keys instead of list of dictionaries.